packages/eslint-plugin/docs/rules/i18n-hyphenated-range.md

```diff
@@ -0,0 +1,23 @@
+# Disallow using a hyphenated numerical ranges in translatable strings (i18n-hyphen-range)
+
+Using hyphenated numerical ranges in translatable strings is incompatible with Android linting rules and can lead to translation pipeline issues. See [Android Lint: TypographyDashes](https://android.googlesource.com/platform/tools/base/+/master/lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/TypographyDetector.java#58) for more information.
+
+## Rule details
+
+Examples of **incorrect** code for this rule:
+
+```js
+__( '1-10' );
+__( '1 - 10' );
+__( '1  -  10' );
+```
+
+Examples of **correct** code for this rule:
+
+```js
+__( '-10' );
+__( '1–10' ); // Note the use of 'en' dash (U+2013) vs hyphen (U+002D)
+__( '1—10' ); // Note use of 'em' dash (U+2014) vs hyphen (U+002D)
+__( '1\u201310' );
+__( 'Text - with a hyphen - is fine.' );
```

Comment:
I noticed that the following string `1 -2` is marked as a lint error, however, is a correct case ([reference](https://android.googlesource.com/platform/tools/base/+/master/lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/TypographyDetector.java#270)).

Reply:
nice, I'll add that 👍

Reply:
added in https://github.com/WordPress/gutenberg/pull/38225/commits/cdf350f8f18d81183fce150dac1a17e371c42b0e

Reply:
fixed in https://github.com/WordPress/gutenberg/pull/38225/commits/a30012e88d68ed06e6fd71183cb78e8635bcf587